utility/drivers.cpp

```diff
@@ -199,78 +159,190 @@ void l3gd20h_getRawTemperature(int8_t *pRawTemperature)
 {
   if(NULL != pRawTemperature)
   {
-    _readFromRegAddr(L3GD20_ADDRESS, L3GD20_GYRO_REGISTER_OUT_TEMP, pRawTemperature, sizeof(*pRawTemperature));
+    readFromRegAddr(L3GD20_ADDRESS, L3GD20_GYRO_REGISTER_OUT_TEMP, pRawTemperature, sizeof(*pRawTemperature));
   }
 }
 
 /*
  * LSM303 Accel + Mag Sensor
+ *
+ * LSM303_D Datasheet:
+ * http://www.st.com/web/en/resource/technical/document/datasheet/DM00057547.pdf
+ *
+ * LSM303_DLHC Datasheet:
+ * https://www.adafruit.com/datasheets/LSM303DLHC.PDF
  */
 LSM303 lsm;
 
-void _lsm303_set_sdk_options() {
-  // update accelerometer gain
+// LSM303 Accelerometer Configurations
+void _lsm303_accel_config(lsm303_accel_gain_e gGain) {
+  uint8_t gain;
+  _lsm303_d_accel_config.gain = gGain;
+
+  // LSM303 D (Spaceboard variant)
   if (lsm.getDeviceType() == LSM303::device_D) {
-    lsm.writeReg(LSM303::CTRL2, LSM303_ACCEL_GAIN8G);
-    _lsm303_d_accel_config.gain = LSM303_ACCEL_GAIN8G;
-  } else {
-    // Ctrl register FS1/FS0 for gain
-    // 00 +/- 2
-    // 01 +/- 4
-    // 10 +/- 8
-    // 11 +/- 16
-    // BDU BLE FS1 FS0 HR 0 0 SIM
-    // 0    0   0   0   1 0 0  0
-    // 0b00011000 = 0x18 (High res, +/- 4g)
-    lsm.writeAccReg(LSM303::CTRL_REG4_A, 0x38);
-    _lsm303_d_accel_config.gain = LSM303_ACCEL_GAIN16G;
-  }
-
-  // update mag scaling
+    // Adjustable gain:
+    //   AFS = 0b00000000 (+/-  2 g full scale) = 0x00
+    //   AFS = 0b00001000 (+/-  4 g full scale) = 0x08
+    //   AFS = 0b00010000 (+/-  6 g full scale) = 0x10
+    //   AFS = 0b00011000 (+/-  8 g full scale) = 0x18 (Default)
+    //   AFS = 0b00100000 (+/- 16 g full scale) = 0x20
+    switch (gGain) {
+      case LSM303_ACCEL_GAIN2G:
+        gain = 0x00;
+        break;
+      case LSM303_ACCEL_GAIN4G:
+        gain = 0x08;
+        break;
+      case LSM303_ACCEL_GAIN6G:
+        gain = 0x10;
+        break;
+      case LSM303_ACCEL_GAIN8G:
+        gain = 0x18;
+        break;
+      case LSM303_ACCEL_GAIN16G:
+        gain = 0x20;
+        break;
+    }
+    lsm.writeReg(LSM303::CTRL2, gain);
+
+    // 0x57 = 0b01010111
+    // AODR = 0101 (50 Hz ODR); AZEN = AYEN = AXEN = 1 (all axes enabled)
+    lsm.writeReg(LSM303::CTRL1, 0x57);
+  }
+  else // LSM303 DLHC (Breakout Space Kit variant)
+  {
+    // Adustable gain:
+    //   FS = 0b00001000 (+/-  2 g full scale; high resolution enable) = 0x08
+    //   FS = 0b00011000 (+/-  4 g full scale; high resolution enable) = 0x18
+    //   FS = 0b00101000 (+/-  8 g full scale; high resolution enable) = 0x28 (Default)
+    //   FS = 0b00111000 (+/- 16 g full scale; high resolution enable) = 0x38
+    switch (gGain) {
+      case LSM303_ACCEL_GAIN2G:
+        gain = 0x08;
+        break;
+      case LSM303_ACCEL_GAIN4G:
+        gain = 0x18;
+        break;
+      case LSM303_ACCEL_GAIN6G:
+        _lsm303_d_accel_config.gain = LSM303_ACCEL_GAIN8G;
```

Comment:
if you meant to roll through here, call it out explicitly in a comment. I'm not sure I understand why this logic is here?